chapter_1/kmeans_hardness_d2_k2.tex

```diff
@@ -0,0 +1,150 @@
+form a grid, $H_{l,n}$. We define $w \overset{\text{\tiny d}}{=} \text{poly}(l, n)$ where $w$ is
+``large enough'' relative to $l,n$, and define the distances and weights in
+$H_{l,n}$ in terms of $w$ as well. We then argue that there are ``nice'' ways
+to cluster these points, such that the cost incurred is ``small'' relative to
+other clusterings. The minimal cost is $L_1-l\alpha$, where $L_1$ and $\alpha$
+are defined in terms of $l,n$.
+
+Then we add some more points $X$ to $H_{l,n}$ to create $G_{l,n}\cup X$. These
+new points are related to the elements in $S_i$ for $S_i\in\cC$. We then argue
+that iff the X3C instance is satisfiable, these new points can be clustered so
+that the total cost increases by an amount $L_2+(l-n)\alpha$ ($L_2$ defined in
+terms of $l,n$). Therefore our cost limit in the decision problem will be $L_1+L_2-n\alpha$.
+
+We will explain the above more rigorously in the next section. Given this construction, the $k$-means instance $G_{l,n}$ will contain exactly $l(6n+3)+(l-1)9n$ points, for which $k=\Theta(n^\gamma)$ for some $0<\gamma<1$ (according to the magic quantity for $k$ mentioned above; also note that the second $n$ here refers to the number of points in the $k$-means instance, not the $n$ in the 3XC instance). There is a simple reduction to the more general $k=\Theta(n^\epsilon)$ for $\epsilon>0$.\\
+
+\begin{theorem}
+    Suppose Theorem 1 is true. Then the $k$-means clustering problem is also NP-hard for $k=\Theta(n^\epsilon)$ for any $\epsilon>0$.
+\end{theorem}
+
+We reproduce the proof almost verbatim from the paper. Fix $\epsilon>0$, and take a hard instance with $n$ points and $k$ centers, with $k=\Theta(n^\gamma)$ as in Theorem 1.
+
+Suppose $\gamma<\epsilon$. Add $n^\epsilon$ points very far from the original
+instance and very far from each other, and $n^\epsilon$ centers. The optimal
+solution will use the new centers to cluster the new points, and the optimum
+in the original instance will not change. Therefore this is a hard instance with
+$m \overset{\text{\tiny d}}{=} n+n^\epsilon=\Theta(n)$ points and
+$k' \overset{\text{\tiny d}}{=} k+n^\epsilon=\Theta(n^\epsilon)$ centers.
+
+Suppose $\gamma>\epsilon$. Add $n^{\gamma/\epsilon}$ points very close to each
+other and far from the original instance, and 1 center. The optimal solution
+will use the new center to cluster the new points, and the optimum in the
+original instance will not change. Therefore this is a hard instance with
+$m \overset{\text{\tiny d}}{=} n+n^{\gamma/\epsilon}=\Theta(n^{\gamma/\epsilon})$ points and
+$k' \overset{\text{\tiny d}}{=} k+1=\Theta(n^\gamma)=\Theta(m^\epsilon)$ centers.\qed
+
+\subsection{Walking through the proof}
+
+Now we will present the paper's proof of Theorem 1. Instead of re-typing everything, we will instead try to expand on explanations and provide some more intuition, and refer the reader to the paper for the technical details.
```

Comment:
This whole section is a bit hard to read without having read the paper before hand. Giving a summary/intuition and omitting technical details is totally fine (and ideal). It makes sense to refer readers to the paper for technical details. However, try to make the intuition/summary self contained. Here i dont know what definitions 4 and lemma 6 are. Nor do i know precisely what lemma 7 is, etc. The "let's read up until" comments make this more of a reading along guide than a self contained summary. Try to edit it a bit to make it closer to the latter (although again referring readers to the paper for technical details is what we want to do).